ruby-service/app/services/authentication_service.rb

```
@@ -0,0 +1,87 @@
+    current_time = Time.now
+    recent_attempts = (@failed_attempts[username] || []).select do |time|
+      current_time - time < FAILED_ATTEMPT_WINDOW_SECONDS
+    end
+    recent_attempts << current_time
+    @failed_attempts[username] = recent_attempts
+
+    block_user(username) if recent_attempts.length >= MAX_FAILED_ATTEMPTS
+  end
+
+  def block_user(username)
+  end
```

RuboCop (1.88.2)

[convention] Put empty method definitions on a single line.

(Style/EmptyMethod)